package/kedro_viz/models/flowchart.py

```diff
@@ -278,24 +304,36 @@ def has_metadata(self) -> bool:
         return self.kedro_obj is not None
 
 
-@dataclass
-class GraphNodeMetadata(abc.ABC):
+class GraphNodeMetadata(BaseModel, abc.ABC):
     """Represent a graph node's metadata"""
 
 
-@dataclass
 class TaskNode(GraphNode):
-    """Represent a graph node of type TASK"""
+    """Represent a graph node of type TASK
+
+    Raises:
+        AssertionError: If kedro_obj is not supplied during instantiation
+    """
 
-    modular_pipelines: List[str] = field(init=False)
-    parameters: Dict = field(init=False, default_factory=dict)
+    parameters: Dict = Field(
+        {}, description="A dictionary of parameter values for the task node"
+    )
+
+    # The type for Task node
     type: str = GraphNodeType.TASK.value
 
-    def __post_init__(self):
-        self.namespace = self.kedro_obj.namespace
+    @root_validator(pre=True)
+    def check_kedro_obj_exists(cls, values):
+        assert "kedro_obj" in values
+        return values
```

Comment:
Can you use `@validator` instead? From my understanding `root_validator` is used for validating entire model data, here it seems that you are interested in a single field.